src/typing_extensions.py

```diff
@@ -111,17 +111,23 @@ def _check_generic(cls, parameters, elen=_marker):
                         f" actual {alen}, expected {elen}")
 
 
+def _is_generic_alias_like(t) -> bool:
+    return hasattr(t, "__args__") and hasattr(t, "__origin__") and hasattr(t, "__parameters__")
```

Comment:
I tried to make an `@runtime_checkable` Protocol but kept running into issues with import order and such.